client/tink_keyset_helper.go

```diff
@@ -119,3 +130,97 @@ func readTinkKeysetFromBytes(data []byte) (*tinkpb.Keyset, error) {
 	}
 	return tinkKeyset, nil
 }
+
+// getTinkKeysetHandleFromKnoxVersionList returns a tink keyset handle that has all tink keys in the
+// received knox version list and a map from tink key IDs to knox version IDs. Since each version is
+// a tink keyset that contains a single key, this func will enumerate all versions in the given version
+// list and do the appending.
+func getTinkKeysetHandleFromKnoxVersionList(
+	knoxVersionList knox.KeyVersionList,
+) (*keyset.Handle, map[uint32]uint64, error) {
+	tinkKeyset := new(tinkpb.Keyset)
+	tinkKeyIDToKnoxVersionID := make(map[uint32]uint64)
+	for _, v := range knoxVersionList {
+		// the data of each version is a tink keyset that contains a single tink key
+		keysetContainsASingleKey, err := readTinkKeysetFromBytes(v.Data)
+		if err != nil {
+			return nil, nil, err
+		}
+		singleKey := keysetContainsASingleKey.Key[0]
+		if v.Status == knox.Primary {
+			tinkKeyset.PrimaryKeyId = singleKey.KeyId
+		}
+		tinkKeyset.Key = append(tinkKeyset.Key, singleKey)
+		tinkKeyIDToKnoxVersionID[singleKey.KeyId] = v.ID
+	}
+	// Convert tink keyset to tink keyset handle. tink doesn't allow transferring cleartext keyset to keyset
+	// handle driectly. Hence, keyset is converted to bytes, then read by package insecurecleartextkeyset
+	bytesBuffer := new(bytes.Buffer)
+	writer := keyset.NewBinaryWriter(bytesBuffer)
+	writer.Write(tinkKeyset)
+	reader := keyset.NewBinaryReader(bytesBuffer)
+	keysetHandle, err := insecurecleartextkeyset.Read(reader)
+	if err != nil {
+		fatalf("cannot get tink keyset handle: %v", err)
+	}
+	return keysetHandle, tinkKeyIDToKnoxVersionID, nil
+}
+
+// getKeysetInfoFromTinkKeysetHandle returns a string representation of the info of the given tink keyset
+// handle. The returned string which does not contain any sensitive key material.
+func getKeysetInfoFromTinkKeysetHandle(
+	keysetHandle *keyset.Handle,
+	tinkKeyIDToKnoxVersionID map[uint32]uint64,
+) (string, error) {
+	// translate the info from the tink build-in function to json format
+	keysetInfo := newJSONTinkKeysetInfo(keysetHandle.KeysetInfo(), tinkKeyIDToKnoxVersionID)
+	keysetInfoForPrint, err := json.MarshalIndent(keysetInfo, "", "  ")
+	if err != nil {
+		return "", err
+	}
+	return string(keysetInfoForPrint), nil
+}
+
+// JSONTinkKeysetInfo translates tink keyset info to JSON format, doesn't contain any actual key material.
+type JSONTinkKeysetInfo struct {
+	PrimaryKeyId uint32                        `json:"primary_key_id"`
+	KeyInfo      []*JSONTinkKeysetInfo_KeyInfo `json:"key_info"`
+}
+
+// JSONTinkKeysetInfo_KeyInfo translates tink key info to JSON format, doesn't contain any actual key material.
+type JSONTinkKeysetInfo_KeyInfo struct {
```

Comment:
JSON in the struct names is not necessary, also I'd suggest merging these structs or coming up with better names to distinguish them.

Reply:
perhaps TinkKeysetInfo and TinkKeyInfo.